Tests/NIOHTTP2Tests/ConfiguringPipelineAsyncMultiplexerTests.swift

```diff
@@ -291,6 +316,201 @@ final class ConfiguringPipelineAsyncMultiplexerTests: XCTestCase {
+        }
+
+        serverRecorder.receivedFrames.assertFramePayloadsMatch(Array(repeating: ConfiguringPipelineAsyncMultiplexerTests.requestFramePayload, count: requestCount))
+    }
+
+    // `testNegotiatedHTTP1BasicPipelineCommunicates` ensures that a client-server system set up to use async stream abstractions
+    // can communicate successfully when HTTP/1.1 is negotiated.
+    func testNegotiatedHTTP1BasicPipelineCommunicates() async throws {
+        let requestCount = 100
+
+        let _ = try await self.clientChannel.pipeline.addHTTPClientHandlers().map { _ in
+            self.clientChannel.pipeline.addHandlers([HTTP1ClientResponseRecorderHandler(), HTTP1ClientSendability()])
+        }.get()
+
+        let nioProtocolNegotiationHandler = try await self.serverChannel.configureAsyncHTTPServerPipeline(
+            connectionConfiguration: .init(),
+            streamConfiguration: .init()
+        ) { channel in
+            channel.pipeline.addHandlers([HTTP1OKResponder(), HTTP1ServerRequestRecorderHandler()])
+        } http2ChannelConfigurator: { channel in
+            channel.eventLoop.makeSucceededVoidFuture()
+        } http2StreamInitializer: { channel -> EventLoopFuture<Channel> in
+            channel.eventLoop.makeSucceededFuture(channel)
+        }.get()
+
+        // Let's pretend the TLS handler did protocol negotiation for us
+        self.serverChannel.pipeline.fireUserInboundEventTriggered(TLSUserEvent.handshakeCompleted(negotiatedProtocol: "http/1.1"))
+
+        let nioProtocolNegotiationResult = try await nioProtocolNegotiationHandler.protocolNegotiationResult.get()
+
+        try await self.interactInMemory(self.clientChannel, self.serverChannel)
+
+        switch nioProtocolNegotiationResult {
+        case .deferredResult:
+            preconditionFailure("Negotiation result must be ready")
+        case .finished(let negotiationResult):
+            switch negotiationResult {
+            case .http1_1:
+                break
+            case .http2:
+                preconditionFailure("Negotiation result must be http/1.1")
+            }
+        }
+
+        // client
+        for _ in 0 ..< requestCount {
+            // Let's try sending some http/1.1 requests.
+            // we need to put these through a mapping to remove references to `IOData` which isn't Sendable
+            try await self.clientChannel.writeOutbound(HTTP1ClientSendability.RequestPart.head(ConfiguringPipelineAsyncMultiplexerTests.requestHead))
+            try await self.clientChannel.writeAndFlush(HTTP1ClientSendability.RequestPart.end(nil)).get()
+            try await self.interactInMemory(self.clientChannel, self.serverChannel)
+        }
+
+        // check expectations
+        let clientRecorder = try await self.clientChannel.pipeline.handler(type: HTTP1ClientResponseRecorderHandler.self).get()
+        let serverRecorder = try await self.serverChannel.pipeline.handler(type: HTTP1ServerRequestRecorderHandler.self).get()
```

Comment:
This is not safe. Those two types are not `Sendable`

Reply:
I've made these recorder types `Sendable`